### ISSUE TITLE
AAR Iris Tracker problem with focallength  and stuck on blackscreen

### ISSUE THREAD

**Issue by onuralpszr** [google/mediapipe#3230]
<em>Please make sure that this is a [solution](https://google.github.io/mediapipe/solutions/solutions.html) issue.<em>

**System information** (Please provide as much relevant information as possible)
- Have I written custom code (as opposed to using a stock example script provided in Mediapipe): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04, Android 11, iOS 14.4): Linux Fedora 35 / Android 12
- [MediaPipe version](https://github.com/google/mediapipe/releases): 0.8.9
- Bazel version: 4.2.1 (managed by bazelisk)
- Solution (e.g. FaceMesh, Pose, Holistic): Iris Tracking
- Programming Language and version : Kotlin

**Describe the expected behavior:**
I was trying to get run current Iris Tracking
https://github.com/google/mediapipe/blob/master/mediapipe/examples/android/src/java/com/google/mediapipe/apps/iristrackinggpu/MainActivity.java

Example in kotlin with AAR solution. In first I was getting focal lenth error because camera needs to start before graph and get length then go to graph, I tried to handle that via global variable and get to value, I got the value then, If I try to open app , it is just close without proper logcat. After I scroll bit more 

E/libEGL: validate_display:547 error 3008 (EGL_BAD_DISPLAY)

And in end I notice that 

A/libc: Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0 in tid 28784 (mediapipe/28784), pid 28615 (face_iris)

**Standalone code you may have used to try to get what you need :**

```kotlin

package com.os.mp_iris_face

import android.graphics.SurfaceTexture
import android.os.Bundle
import android.util.Log
import android.util.Size
import android.view.SurfaceHolder
import android.view.SurfaceView
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import com.google.mediapipe.components.CameraHelper.CameraFacing
import com.google.mediapipe.components.CameraXPreviewHelper
import com.google.mediapipe.components.ExternalTextureConverter
import com.google.mediapipe.components.FrameProcessor
import com.google.mediapipe.components.PermissionHelper
import com.google.mediapipe.formats.proto.LandmarkProto.NormalizedLandmarkList
import com.google.mediapipe.framework.AndroidAssetUtil
import com.google.mediapipe.framework.Packet
import com.google.mediapipe.framework.PacketGetter
import com.google.mediapipe.glutil.EglManager


class MainActivity : AppCompatActivity() {
    private val BINARY_GRAPH_NAME = "iris_tracking_gpu.binarypb"
    private val INPUT_VIDEO_STREAM_NAME = "input_video"
    private val OUTPUT_VIDEO_STREAM_NAME = "output_video"
    private val FOCAL_LENGTH_STREAM_NAME = "focal_length_pixel"
    private val OUTPUT_LANDMARKS_STREAM_NAME = "face_landmarks_with_iris"
    private val CAMERA_FACING = CameraFacing.FRONT
    private val FLIP_FRAMES_VERTICALLY = true
    private val NUM_BUFFERS = 2
    private var haveAddedSidePackets = false
    private val TAG: String = "IrisMainAct"

    var focallen  = 1.4E-45

    companion object {

        init {
            
            System.loadLibrary("mediapipe_jni")
            System.loadLibrary("opencv_java4")
        }
    }

    
    private var previewFrameTexture: SurfaceTexture? = null
    private var previewDisplayView: SurfaceView? = null
    private var eglManager: EglManager? = null
    private var processor: FrameProcessor? = null
    private var converter: ExternalTextureConverter? = null
    private var cameraHelper: CameraXPreviewHelper? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        Log.wtf(TAG, "onCreate method running")
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        previewDisplayView = SurfaceView(this)
        setupPreviewDisplayView()
        
        
        AndroidAssetUtil.initializeNativeAssetManager(this)
        eglManager = EglManager(null)
        processor = FrameProcessor(
            this,
            eglManager!!.nativeContext,
            BINARY_GRAPH_NAME,
            INPUT_VIDEO_STREAM_NAME,
            OUTPUT_VIDEO_STREAM_NAME
        )
        processor!!.videoSurfaceOutput.setFlipY(FLIP_FRAMES_VERTICALLY)
        PermissionHelper.checkAndRequestCameraPermissions(this)


        if (!haveAddedSidePackets) {
            val focalLength = focallen
            Log.wtf(TAG,"Focal len value")
            Log.wtf(TAG,focalLength.toString())
            if (focalLength != Double.MIN_VALUE) {
                val focalLengthSidePacket = processor!!.packetCreator.createFloat32(focalLength.toFloat())
                val inputSidePackets: MutableMap<String, Packet> = HashMap()
                inputSidePackets[FOCAL_LENGTH_STREAM_NAME] = focalLengthSidePacket
                processor!!.setInputSidePackets(inputSidePackets)
            }
            haveAddedSidePackets = true
        }



        processor!!.addPacketCallback(
            OUTPUT_LANDMARKS_STREAM_NAME

        ) { packet: Packet ->
            Log.wtf(TAG, "Received iris landmarks packet.")
            val landmarksRaw = PacketGetter.getProtoBytes(packet)
            val landmarks: NormalizedLandmarkList = NormalizedLandmarkList.parseFrom(landmarksRaw)

            Log.wtf(
                TAG,
                "[TS:"
                        + packet.timestamp
                        + "] "
                        + getLandmarksDebugString(landmarks)
            )
        }


    }


    override fun onResume() {
        super.onResume()
        if (PermissionHelper.cameraPermissionsGranted(this)) {
            startCamera()
        }
        converter = ExternalTextureConverter(
            eglManager!!.context, NUM_BUFFERS
        )
        converter!!.setFlipY(FLIP_FRAMES_VERTICALLY)
        converter!!.setConsumer(processor)

    }


    override fun onPause() {
        super.onPause()
        converter!!.close()

        
        previewDisplayView!!.visibility = View.GONE
    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<String>,
        grantResults: IntArray
    ) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        PermissionHelper.onRequestPermissionsResult(requestCode, permissions, grantResults)
    }


    private fun startCamera() {
        cameraHelper = CameraXPreviewHelper()
        cameraHelper!!.setOnCameraStartedListener { surfaceTexture: SurfaceTexture? ->
            previewFrameTexture = surfaceTexture
            
            
            //previewDisplayView!!.visibility = View.VISIBLE
            previewDisplayView!!.visibility = View.VISIBLE
        }
        cameraHelper!!.startCamera(this, CAMERA_FACING,  /*surfaceTexture=*/null)
        focallen = cameraHelper!!.focalLengthPixels.toDouble()
        Log.wtf(TAG, "CameraX_Focal_Len $focallen")

    }

    private fun setupPreviewDisplayView() {
        previewDisplayView!!.visibility = View.GONE
        val viewGroup = findViewById<ViewGroup>(R.id.preview_display_layout)
        viewGroup.addView(previewDisplayView)
        previewDisplayView!!
            .getHolder()
            .addCallback(
                object : SurfaceHolder.Callback {
                    override fun surfaceCreated(holder: SurfaceHolder) {
                        processor!!.videoSurfaceOutput.setSurface(holder.surface)
                    }

                    override fun surfaceChanged(
                        holder: SurfaceHolder,
                        format: Int,
                        width: Int,
                        height: Int
                    ) {
                        
                        
                        
                        val viewSize = Size(width, height)
                        val displaySize = cameraHelper!!.computeDisplaySizeFromViewSize(viewSize)

                        
                        
                        
                        converter!!.setSurfaceTextureAndAttachToGLContext(
                            previewFrameTexture, displaySize.width, displaySize.height
                        )
                    }

                    override fun surfaceDestroyed(holder: SurfaceHolder) {
                        processor!!.videoSurfaceOutput.setSurface(null)
                    }
                })
    }

    private fun getLandmarksDebugString(landmarks: NormalizedLandmarkList): String? {
        var landmarkIndex = 0
        var landmarksString = ""
        for (landmark in landmarks.landmarkList) {
            landmarksString += """		Landmark[$landmarkIndex]: (${landmark.x}, ${landmark.y}, ${landmark.z})
"""
            ++landmarkIndex
        }
        return landmarksString
    }

}

```

**Other info / Complete Logs :**

```
E/AndroidRuntime: FATAL EXCEPTION: ExternalTextureConverter
    Process: com.os.mp_iris_face, PID: 1346
    java.lang.RuntimeException: Error during attachToGLContext (see logcat for details)
        at android.graphics.SurfaceTexture.attachToGLContext(SurfaceTexture.java:296)
        at com.google.mediapipe.components.ExternalTextureConverter$RenderThread.setSurfaceTextureAndAttachToGLContext(ExternalTextureConverter.java:310)
        at com.google.mediapipe.components.ExternalTextureConverter.lambda$setSurfaceTextureAndAttachToGLContext$2$com-google-mediapipe-components-ExternalTextureConverter(ExternalTextureConverter.java:197)
        at com.google.mediapipe.components.ExternalTextureConverter$$ExternalSyntheticLambda1.run(Unknown Source:8)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:226)
        at android.os.Looper.loop(Looper.java:313)
        at com.google.mediapipe.glutil.GlThread.run(GlThread.java:141)
        
        
        
        
        E/native: E20220404 14:53:42.393757  3176 graph.cc:480] INVALID_ARGUMENT: ValidateRequiredSidePackets failed to validate: 
    ; Side packet "focal_length_pixel" is required but was not provided.
D/GlThread: Stopping GL thread ExternalTextureConverter
E/AndroidRuntime: FATAL EXCEPTION: ExternalTextureConverter
    Process: com.os.mp_iris_face, PID: 3109
    com.google.mediapipe.framework.MediaPipeException: invalid argument: ValidateRequiredSidePackets failed to validate: 
    ; Side packet "focal_length_pixel" is required but was not provided.
        at com.google.mediapipe.framework.Graph.nativeStartRunningGraph(Native Method)
        at com.google.mediapipe.framework.Graph.startRunningGraph(Graph.java:324)
        at com.google.mediapipe.components.FrameProcessor.startGraph(FrameProcessor.java:553)
        at com.google.mediapipe.components.FrameProcessor.maybeAcceptNewFrame(FrameProcessor.java:421)
        at com.google.mediapipe.components.FrameProcessor.onNewFrame(FrameProcessor.java:439)
        at com.google.mediapipe.components.ExternalTextureConverter$RenderThread.renderNext(ExternalTextureConverter.java:403)
        at com.google.mediapipe.components.ExternalTextureConverter$RenderThread.lambda$onFrameAvailable$0$com-google-mediapipe-components-ExternalTextureConverter$RenderThread(ExternalTextureConverter.java:338)
        at com.google.mediapipe.components.ExternalTextureConverter$RenderThread$$ExternalSyntheticLambda1.run(Unknown Source:4)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:226)
        at android.os.Looper.loop(Looper.java:313)
        at com.google.mediapipe.glutil.GlThread.run(GlThread.java:141)
W/System: A resource failed to call close. 

```

**Comment by thunderbirdtr** [comment]
Bazel BUILD file content

```
load("//mediapipe/java/com/google/mediapipe:mediapipe_aar.bzl", "mediapipe_aar")

mediapipe_aar(
    name = "mediapipe_face_detection",
    calculators = ["//mediapipe/graphs/iris_tracking:iris_tracking_gpu_deps"],
)
```

**Comment by thunderbirdtr** [comment]
Quick update : I recompile mediapipe iris part with "opencv3" in order to see any difference but still getting same error. at lest "focal" one. 

I changed to behavior and I set focal value hardcoded, and I was able to "not see" the focal error but APK is still crashing without giving me proper error. Any idea and feedback would be really helpful. I also notice  that this problem happen before and there is a corresponding bug link , I follow that steps but didn't reach to result. Right now  I'm trying to replicate the code with java instead of kotlin in order to achieve working solution or  see same error.
